src/com/microsoft/azure/hdinsight/serverexplore/UI/AddNewClusterFrom.java

```diff
@@ -0,0 +1,186 @@
+package com.microsoft.azure.hdinsight.serverexplore.UI;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.microsoft.azure.hdinsight.common.HDInsightHelper;
+import com.microsoft.azure.hdinsight.common.StringHelper;
+import com.microsoft.azure.hdinsight.sdk.cluster.HDInsightAdditionalClusterDetail;
+import com.microsoft.azure.hdinsight.sdk.storage.StorageAccount;
+import com.microsoft.azure.hdinsight.serverexplore.hdinsightnode.HDInsightRootModule;
+import com.microsoft.azure.hdinsight.spark.common.AddNewClusterImpl;
+import com.microsoft.azure.hdinsight.spark.common.StorageAccountResolveException;
+import com.microsoft.azure.hdinsight.spark.common.UnAuthorizedException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Created by ltian on 12/17/2015.
+ */
+public class AddNewClusterFrom extends DialogWrapper {
+
+    private Project project;
+    private String clusterName;
+    private String userName;
+    private String password;
+    private String errorMessage;
+
+    private boolean isCarryOnNextStep;
+    private List<StorageAccount> storageAccounts = null;
+
+    private JPanel addNewClusterPanel;
+    private JTextField clusterNameFiled;
+    private JTextField userNameField;
+    private JPasswordField passwordField;
+    private JTextField errorMessageField;
+    private JPanel buttonPanel;
+    private JButton Okbutton;
+    private JButton cancelButton;
+
+    private static final String DEFAULT_CLUSTER_ENDPOINT = "https://%s.azurehdinsight.net";

```

Comment:
I remember DEFAULT_CLUSTER_ENDPOINT was referenced  more more than once, try to put it in a common string (HDInsight sdk common is a good place)